src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,206 @@
+        }
+        deleteDriversByCar(car);
+        addDriversToCar(car);
+        return car;
+    }
+
+    private boolean deleteDriversByCar(Car car) {
+        String deleteAllDriversQuery = "DELETE FROM cars_drivers WHERE car_id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                    PreparedStatement statement
+                            = connection.prepareStatement(deleteAllDriversQuery)) {
+            statement.setLong(1, car.getId());
+            return statement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException(
+                    "Can't deleted drivers by car " + car, e);
+        }
+    }
+
+    private Car addDriversToCar(Car car) {
+        String insertDriversQuery =
+                "INSERT INTO cars_drivers (car_id, driver_id) VALUES (?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                    PreparedStatement statement
+                            = connection.prepareStatement(
+                                    insertDriversQuery)) {
+            statement.setLong(1, car.getId());
+            for (Driver driver : car.getDrivers()) {
+                statement.setLong(2, driver.getId());
+                statement.executeUpdate();
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't insert drivers to car " + car, e);
+        }
+        return car;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE cars SET is_deleted = TRUE WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                    PreparedStatement deleteDriverStatement
+                                = connection.prepareStatement(query)) {
+            deleteDriverStatement.setLong(1, id);
+            return deleteDriverStatement.executeUpdate() > 0;
+        } catch (SQLException throwable) {
+            throw new DataProcessingException("Couldn't delete car with id " + id, throwable);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String selectDrivers = "SELECT c.id, c.title, c.manufacturer_id"
+                + " FROM cars c JOIN cars_drivers cd ON "
+                + "cd.car_id = c.id"
+                + " WHERE cd.car_id = ? AND c.is_deleted = false";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                    PreparedStatement selectStatement
+                        = connection.prepareStatement(selectDrivers)) {
+            selectStatement.setLong(1, driverId);
+            ResultSet resultSet = selectStatement.executeQuery();
+            Car car = new Car();
+            while (resultSet.next()) {
+                Long id = resultSet.getObject("id", Long.class);
+                car.setDrivers(getDriversById(id, connection));
```

Comment:
let's call this method after `catch` block for each of the cars in our `cars` list
the purpose is not to open a connection inside another connection